### PR TITLE
[BEAM-1504] Parameterized tests for standard coders in Python SDK

### PR DESCRIPTION
**Pull request by vikkyrk**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

`python setup.py test -s apache_beam.coders.standard_coders_test`

...

`test_standard_coder_0_bytes (apache_beam.coders.standard_coders_test.StandardCodersTest) ... ok
test_standard_coder_1_bytes (apache_beam.coders.standard_coders_test.StandardCodersTest) ... ok
test_standard_coder_2_varint (apache_beam.coders.standard_coders_test.StandardCodersTest) ... ok
test_standard_coder_3_kv (apache_beam.coders.standard_coders_test.StandardCodersTest) ... ok
test_standard_coder_4_kv (apache_beam.coders.standard_coders_test.StandardCodersTest) ... ok
test_standard_coder_5_kv (apache_beam.coders.standard_coders_test.StandardCodersTest) ... ok
test_standard_coder_6_intervalwindow (apache_beam.coders.standard_coders_test.StandardCodersTest) ... ok
test_standard_coder_7_stream (apache_beam.coders.standard_coders_test.StandardCodersTest) ... ok
test_standard_coder_8_stream (apache_beam.coders.standard_coders_test.StandardCodersTest) ... ok
`

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by vikkyrk** [comment]
R: @aaltay 
CC: @dhalperi @robertwb

**Comment by dhalperi** [comment]
Nice, thanks.

**Comment by vikkyrk** [comment]
R: @jasonkuster for pom changes.

File: sdks/python/setup.py [hunk 1]
```diff
@@ -100,6 +100,7 @@ def get_version():
 
 
 REQUIRED_TEST_PACKAGES = [
+    'nose-parameterized>=0.5.0',
```
Comment (aaltay) [inline sdks/python/setup.py]: Could you upper bound this with `<0.6.0`

File: sdks/python/apache_beam/coders/standard_coders_test.py [hunk 2]
```diff
@@ -30,6 +30,20 @@
 from apache_beam.utils.timestamp import Timestamp
 from apache_beam.transforms.window import IntervalWindow
 
+from nose_parameterized import parameterized
+
+STANDARD_CODERS_YAML = os.path.join(
+    os.path.dirname(__file__), '..', 'tests', 'data', 'standard_coders.yaml')
+
+
+def data(test_yaml):
```
Comment (aaltay) [inline sdks/python/apache_beam/coders/standard_coders_test.py]: Could you add a comment about what this is doing and use a less generic name for the function.

**Comment by vikkyrk** [comment]
Addressed your comments, and yes it works with python setup.py test, tox etc.

File: sdks/python/setup.py [hunk 3]
```diff
@@ -100,6 +100,7 @@ def get_version():
 
 
 REQUIRED_TEST_PACKAGES = [
+    'nose-parameterized>=0.5.0, <0.6.0',
```
Comment (aaltay) [inline sdks/python/setup.py]: Could you remove the space between `,` and `<`?

Comment (vikkyrk) [inline sdks/python/setup.py]: Done

**Pull request closed without merging**